superset/config.py

```diff
@@ -1293,15 +1294,16 @@ def EMAIL_HEADER_MUTATOR(  # pylint: disable=invalid-name,unused-argument
 }
 
 # It is possible to customize which tables and roles are featured in the RLS
-# dropdown. When set, this dict is assigned to `add_form_query_rel_fields` and
-# `edit_form_query_rel_fields` on `RowLevelSecurityFiltersModelView`. Example:
+# dropdown. When set, this dict is assigned to `filter_rel_fields`
+# on `RLSRestApi`. Example:
 #
 # from flask_appbuilder.models.sqla import filters
-# RLS_FORM_QUERY_REL_FIELDS = {
-#     "roles": [["name", filters.FilterStartsWith, "RlsRole"]]
-#     "tables": [["table_name", filters.FilterContains, "rls"]]
+
+# RLS_FILTER_RELATED_FIELDS = {
+#     "tables": [["table_name", filters.FilterStartsWith, "birth"]],
+#     "roles": [["name", filters.FilterContains, "Admin"]]
 # }
-RLS_FORM_QUERY_REL_FIELDS: Optional[Dict[str, List[List[Any]]]] = None
+RLS_FILTER_RELATED_FIELDS: Dict[str, BaseFilter] = {}
```

Comment:
seems like there's no difference between the old `RLS_FORM_QUERY_REL_FIELDS` and `RLS_FILTER_RELATED_FIELDS` but the type annotation does not seem correct now. We can probably not break compatibility here and add a deprecation note on `RLS_FORM_QUERY_REL_FIELDS`

Reply:
type annotation for `base_related_field_filters` is `Dict[str, BaseFilter]` in `BaseSupersetModelRestApi` base class, so I used the same.
am I missing something?

Reply:
also how to keep the compatibility?
keep both flags in config and in the do like below?
`
if app.config["RLS_FORM_QUERY_REL_FIELDS"]:
  base_related_field_filters = app.config["RLS_FORM_QUERY_REL_FIELDS"]
  logger.warning()
elif app.config["RLS_FILTER_RELATED_FIELDS"]:
  base_related_field_filters = app.config["RLS_FILTER_RELATED_FIELDS"]
`

Reply:
@dpgaspar